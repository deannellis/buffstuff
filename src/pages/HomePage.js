import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter, Link} from "react-router-dom";
import moment from 'moment';

import { startWorkout } from '../actions/inProgressWorkout';
import { setTargetMacros, updateMacro, setCurrentDate, resetCurrent } from '../actions/macros';
import Dashboard from '../components/Dashboard';
import Button from '../components/Button';
import SideNav from '../components/SideNav';

class HomePage extends Component {
    constructor(props) {
        super(props);
        this.state = {
            today: moment()
        }
    }

    onStartWorkout = () => {
        this.props.dispatch(startWorkout(this.props.liftVariant));
        this.props.history.push('/workout');
    }

    onSetMacros = macros => { this.props.dispatch(setTargetMacros(macros)) }

    onUpdateMacro = update => {
        this.props.dispatch(updateMacro(update))
    }

    componentDidMount() {
        const { dateObject } = this.props.macros;
        const storedDate = dateObject._d
        const { today } = this.state;
        if(dateObject === null || dateObject === 0) {
            console.log('SETTING DATE DUE TO NULL OR 0 VALUE');
            this.props.dispatch(setCurrentDate({ currentDate: today }));
        }else if(!(moment().isSame(storedDate, 'day'))) {
            console.log('RESETING CURRENT MACROS BASED ON OUTDATED DATE');
            this.props.dispatch(resetCurrent());
            this.props.dispatch(setCurrentDate({ currentDate: today }));
        }
    }

    render() { 
        if(Object.entries(this.props.masterWeights).length === 0 && this.props.masterWeights.constructor === Object) {
            return (
                <>
                    <h2>Welcome to flext!</h2>
                    <p>Let's begin by entering your starting weights</p>
                    <Link to="/onboarding">
                        <Button variant="primary">
                            Enter Weights
                        </Button>
                    </Link>
                </>
            );
        }
        return (
            <div className="page--with-side-nav">
                <SideNav path={this.props.match.path} />
                <Dashboard
                    liftVariant={this.props.liftVariant}
                    masterWeights={this.props.masterWeights} 
                    workouts={this.props.workouts}
                    onStartWorkout={this.onStartWorkout}
                    macros={this.props.macros}
                    onSetMacros={this.onSetMacros}
                    onUpdateMacro={this.onUpdateMacro}
                />
            </div>
        );
    }
}

const mapStateToProps = state => {
    return {
        masterWeights: state.masterWeights,
        liftVariant: state.liftVariant,
        workouts: state.workouts,
        macros: state.macros,
    };
}
 
export default withRouter(connect(mapStateToProps)(HomePage));