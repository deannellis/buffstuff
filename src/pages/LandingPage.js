import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import { startLogin } from '../actions/auth';
import Button from '../components/Button';
import AppFooter from '../components/AppFooter';

const LoginPage = ({ startLogin }) => {
	return (
		<div className="landing-page">
			<div className="landing-page__hero">
				<div>
					<h1>flext takes the guess work out of weight lifting</h1>
					<Button clickHandler={startLogin()}>
						<img
							src="./images/64px-Google__G__Logo.webp"
							alt="Google mark"
							height="32px"
						/>
						Log in with Google
					</Button>
				</div>
				<div className="landing-page__hero-image">
					<img
						src="./images/landingIllustration.png"
						alt="illustration of weightlifter"
					/>
				</div>
			</div>
			<div className="landing-page__section-2">
				<div>
					<img
						src="./images/workoutComponents.png"
						alt="Components from within the application"
					/>
				</div>
				<div>
					<h1>
						flext knows what your next workout looks like so you don&apos;t have
						to
					</h1>
				</div>
			</div>
			<div className="landing-page__section-3">
				<div>
					<h1>See your gains increase as you progress</h1>
				</div>
				<div>
					<img
						src="./images/weightGraph.png"
						alt="Components from within the application"
					/>
				</div>
			</div>
			<div className="landing-page__section-4">
				<div>
					<img
						src="./images/macroComponent.png"
						alt="Components from within the application"
					/>
				</div>
				<div>
					<h1>Track your macros to optimize your gains</h1>
				</div>
			</div>
			<AppFooter />
		</div>
	);
};

const mapDispatchToProps = (dispatch) => ({
	startLogin: () => dispatch(startLogin),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
