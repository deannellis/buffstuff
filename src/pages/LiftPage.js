import React, { Component } from 'react';
import { withRouter } from "react-router";
import { connect } from 'react-redux';
import { updateWorkout } from '../actions/inProgressWorkout';
import { getDisplayName, getWarmupWeights, getWeightDistribution } from '../utils/workout';
import Button from '../components/Button';

class LiftPage extends Component {
    constructor(props) {
        super(props);
        this.state = {  }
    }

    completeLift = (result) => {
        const liftResult = {updates: {[this.props.match.params.id]: result}}
        console.log(liftResult);
        this.props.dispatch(updateWorkout(liftResult));
        this.props.history.push('/workout');
    }

    render() { 
        let { id } = this.props.match.params;
        const workWeight = this.props.masterWeights[id];
        const workReps = ['5 reps', '5 reps', 'AMRAP*'];
        const { negatives, ups, weight } = this.props.masterWeights.chinup;
        return (
            <div className="lift-page">
                <div className="lift-page__card">
                    <h2>
                        {getDisplayName(id)}
                    </h2>
                    {id === 'chinup' ? (
                        <div>
                            <h3>Complete</h3>
                            <p className="lift-page__chinup-title">3 sets of</p>
                            <div className="lift-page__chinups">
                                {negatives === 0 ? '' : (
                                    <div className="lift-page__chinup">
                                        <p>{negatives}</p>
                                        <span>negatives</span>
                                    </div>
                                )}
                                {ups === 0 ? '' : (
                                    <div className="lift-page__chinup">
                                        <p>{ups}</p>
                                        <span>chin-ups</span>
                                    </div>
                                )}
                                {weight === 0 ? '' : (
                                    <div className="lift-page__chinup">
                                        <p>{weight}</p>
                                        <span>lbs</span>
                                    </div>
                                )}
                            </div>
                            <h3>Enter your results</h3>
                            <p>Were you able to complete all sets and reps?</p>
                            <Button variant="primary" clickHandler={() => {this.completeLift(1)}}>Yes</Button>
                            <Button clickHandler={() => {this.completeLift(0)}}>No</Button>
                        </div>
                    ) : (
                        <>
                            <div className="lift-page__tables">
                                <div>
                                    <h3>Warm-Up</h3>
                                    {getWarmupWeights(workWeight).map((weight, i) => (
                                        <div key={i} className="lift-page__sets">
                                            <div className="lift-page__set">
                                                <span>Set {i + 1}</span>
                                                <p>5 reps @{weight}lbs</p>
                                            </div>
                                            <div className="lift-page__distributed">
                                                {getWeightDistribution(weight).map((distWeight, i) => (
                                                    <div key={i} className="lift-page__distributed-item">
                                                        <span>
                                                            {i === 0 && 'LEFT'}
                                                            {i === 1 && 'BAR'}
                                                            {i === 2 && 'RIGHT'}
                                                        </span>
                                                        <p>{distWeight}</p>
                                                    </div>
                                                ))}
                                            </div>
                                        </div>
                                    ))}
                                </div>
                                <div>
                                    <h3>Work</h3>
                                    {workReps.map((rep, i) => (
                                        <div key={i} className="lift-page__sets">
                                                <div className="lift-page__set">
                                                    <span>Set {i + 1}</span>
                                                    <p>{rep} @{workWeight}lbs</p>
                                                </div>
                                                <div className="lift-page__distributed">
                                                    {getWeightDistribution(workWeight).map((distWeight, i) => (
                                                        <div key={i} className="lift-page__distributed-item">
                                                            <span>
                                                                {i === 0 && 'LEFT'}
                                                                {i === 1 && 'BAR'}
                                                                {i === 2 && 'RIGHT'}
                                                            </span>
                                                            <p>{distWeight}</p>
                                                        </div>
                                                    ))}
                                                </div>
                                        </div>
                                    ))}
                                </div>
                            </div>
                            <p>*as many reps as possible </p>
                            <h3 className="lift-page__results-title">Enter your results</h3>
                            <p>On your last set how many reps did you complete?</p>
                            <Button clickHandler={() => {this.completeLift(0)}}>Less than 5</Button>
                            <Button clickHandler={() => {this.completeLift(1)}}>Greater than 5</Button>
                            <Button clickHandler={() => {this.completeLift(2)}}>Greater than 10</Button>
                        </>
                    )}
                    
                </div>
            </div>
        );
    }
}
 
const mapStateToProps = state => {
    return {
        masterWeights: state.masterWeights,
    };
}
 
export default withRouter(connect(mapStateToProps)(LiftPage));