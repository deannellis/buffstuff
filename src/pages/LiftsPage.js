import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';

import { setWeight } from '../actions/masterWeights';
import { getDisplayName } from '../utils/workout';
import SideNav from '../components/SideNav';
import Tabs from '../components/Tabs';
import LineGraph from '../components/LineGraph';
import LiftCard from '../components/LiftCard';
import MenuContext from '../context/menu-context';

const lifts = ['bench', 'deadlift', 'overhead', 'row', 'squat'];

export class LiftsPage extends Component {
	constructor(props) {
		super(props);
		this.state = {
			activeTab: 0,
			updateFormIsOpen: false,
			currentLift: lifts[0],
		};
	}

	componentWillUnmount() {
		const { closeMenu } = this.context;
		closeMenu();
	}

	getData = () => {
		const { workouts } = this.props;
		const { activeTab } = this.state;
		const selectedLift = lifts[activeTab];
		const filteredWorkouts = [];
		workouts.forEach((workout) => {
			let workoutWithDate = {};
			if (Object.prototype.hasOwnProperty.call(workout, selectedLift)) {
				workoutWithDate = {
					...workout[selectedLift],
					date: workout.created,
				};
				filteredWorkouts.push(workoutWithDate);
			}
		});
		return filteredWorkouts;
	};

	updateLiftWeight = (update) => {
		const { updateLiftWeight } = this.props;
		updateLiftWeight(update);
	};

	handleKeyPress = (e) => {
		const { toggleMenu } = this.context;
		if (e.key === 'Escape') {
			toggleMenu();
		}
	};

	render() {
		const { menuIsOpen, toggleMenu } = this.context;
		const { match, masterWeights } = this.props;
		const { activeTab, updateFormIsOpen, currentLift } = this.state;
		return (
			<div className="page--with-side-nav">
				<SideNav path={match.path} />
				<div className="lifts-page side-nav__page-content">
					<div
						className={
							menuIsOpen
								? 'side-nav__page-scrim'
								: 'side-nav__page-scrim side-nav__page-scrim--hidden'
						}
						onClick={toggleMenu}
						onKeyPress={this.handleKeyPress}
						role="presentation"
					/>
					<Tabs
						activeIndex={activeTab}
						handleSelect={(i) => {
							this.setState({ activeTab: i, currentLift: lifts[i] });
						}}
						labels={lifts.map((lift) => getDisplayName(lift))}
					>
						<div className="card lifts-page__line-graph">
							<h2>Weight Over Time</h2>
							<LineGraph data={this.getData()} />
						</div>
						<LiftCard
							lift={currentLift}
							weights={masterWeights}
							formIsOpen={updateFormIsOpen}
							toggleForm={() => {
								this.setState((prevState) => ({
									updateFormIsOpen: !prevState.updateFormIsOpen,
								}));
							}}
							updateWeight={this.updateLiftWeight}
						/>
					</Tabs>
				</div>
			</div>
		);
	}
}
LiftsPage.contextType = MenuContext;
LiftsPage.propTypes = {
	match: PropTypes.shape({
		path: PropTypes.string,
	}),
	workouts: PropTypes.arrayOf(PropTypes.object),
	masterWeights: PropTypes.shape({
		bench: 0,
		row: 0,
		squat: 0,
		deadlift: 0,
		overhead: 0,
		chinup: PropTypes.object,
	}),
	updateLiftWeight: PropTypes.func.isRequired,
};
LiftsPage.defaultProps = {
	match: { path: '' },
	workouts: [],
	masterWeights: {
		bench: 0,
		row: 0,
		squat: 0,
		deadlift: 0,
		overhead: 0,
		chinup: {},
	},
};

const mapStateToProps = (state) => ({
	workouts: state.workouts,
	masterWeights: state.masterWeights,
});

const mapDispatchToProps = (dispatch) => ({
	updateLiftWeight: (update) => {
		dispatch(setWeight({ update }));
	},
});

export default withRouter(
	connect(mapStateToProps, mapDispatchToProps)(LiftsPage)
);
