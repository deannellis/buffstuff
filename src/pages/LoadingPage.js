import React from 'react';

const LoadingPage = () => {
	return (
		<div className="loading-page">
			<img alt="loading spinner" src="./images/Spinner-1s-200px.gif" />
		</div>
	);
};

export default LoadingPage;
