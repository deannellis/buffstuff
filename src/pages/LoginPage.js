import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import { startLogin } from '../actions/auth';
import Button from '../components/Button';

const LoginPage = ({ startLogin }) => {
	return (
		<>
			<h1>Welcome to Flext</h1>
			<Button clickHandler={startLogin()}>Login to Flext</Button>
			<button>
				<Link to="/home">To home</Link>
			</button>
		</>
	);
};

const mapDispatchToProps = (dispatch) => ({
	startLogin: () => dispatch(startLogin),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
