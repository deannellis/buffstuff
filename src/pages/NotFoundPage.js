import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => {
	return (
		<div className="not-found-page">
			<div className="card">
				<h1>Sorry, we couldn&apos;t find that page...</h1>
				<Link to="/home">Head back home</Link>
			</div>
		</div>
	);
};

export default NotFoundPage;
