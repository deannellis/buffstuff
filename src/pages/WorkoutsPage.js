import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from "react-router-dom";
import moment from 'moment';

import { startWorkout } from '../actions/inProgressWorkout';
import { getDisplayName } from '../utils/workout';
import SideNav from '../components/SideNav';
import Button from '../components/Button';
import { MenuContext } from '../context/menu-context';

class WorkoutsPage extends Component {
    constructor(props) {
        super(props);
        this.state = {  }
    }
    render() { 
        const onStartWorkout = () => {
            this.props.dispatch(startWorkout(liftVariant));
            this.props.history.push('/workout');
        }
        let { menuIsOpen } = this.context;
    
        return (
            <div className="page--with-side-nav">
                <SideNav path={this.props.match.path} />
            
                <div className="workouts-page side-nav__page-content">
                    <div className={menuIsOpen ? 'side-nav__page-scrim' : 'side-nav__page-scrim side-nav__page-scrim--hidden'}></div>
                    <div className="workouts-page__list">
                        <div className="workouts-page__header">
                            <h1>{this.props.workouts.length} total workouts</h1>
                            <Button variant="primary" clickHandler={onStartWorkout}>Start Next Workout</Button>
                        </div>
                        {this.props.workouts.map((workout, i) => {
                            const workoutKeys = Object.keys(workout);
                            return (
                                <div className="workouts-page__workout card" key={i}>
                                    <p className="workouts-page__workout-number">Workout #{i+1}</p>
                                    <div className="empty-grid-cell"></div>
                                    <div className="workouts-page__date">
                                        <p>{moment(workout.created).format("MMMM, D")}</p>
                                    </div>
                                    {workoutKeys.map(key => {
                                        if(key == 'id' || key == 'created') return
                                        return (
                                            <div className="workouts-page__lift" key={key}>
                                                <p className="workouts-page__lift-name">{getDisplayName(key)}</p>
                                                <p className="workouts-page__result">
                                                    Result:  
                                                    {workout[key].result === 0 && key !== 'chinup' ? ' Last set less than 5 reps' : ''} 
                                                    {workout[key].result === 1 && key !== 'chinup' ? ' Last set greater than 5 reps' : ''} 
                                                    {workout[key].result === 2 && key !== 'chinup' ? ' Last set greater than 10 reps' : ''} 
                                                    {workout[key].result === 0 && key === 'chinup' ? ' Did not complete all reps' : ''} 
                                                    {workout[key].result === 1 && key === 'chinup' ? ' All reps completed!' : ''} 
                                                </p>
                                            </div>
                                        );
                                    })}
                                </div>
                            );
                        })}
                    </div>
                </div>
            </div>
        );
    }
}
WorkoutsPage.contextType = MenuContext;
 
const mapStateToProps = state => {
    return {
        workouts: state.workouts,
        liftVariant: state.liftVariant,
    };
}
 
export default withRouter(connect(mapStateToProps)(WorkoutsPage));