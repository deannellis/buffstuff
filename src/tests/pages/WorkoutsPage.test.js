import React from 'react';
import { shallow } from 'enzyme';

import { WorkoutsPage } from '../../pages/WorkoutsPage';

test('should render Workouts Page without props', () => {
    const wrapper = shallow(<WorkoutsPage />);
    expect(wrapper).toMatchSnapshot();
});
