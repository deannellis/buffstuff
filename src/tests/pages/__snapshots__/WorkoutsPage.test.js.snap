// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Workouts Page with props 1`] = `
<div
  className="page--with-side-nav"
>
  <SideNav
    path=""
  />
  <div
    className="workouts-page side-nav__page-content"
  >
    <div
      className="side-nav__page-scrim side-nav__page-scrim--hidden"
    />
    <div
      className="workouts-page__list"
    >
      <div
        className="workouts-page__header"
      >
        <h1>
          3 total workouts
        </h1>
        <Button
          clickHandler={[Function]}
          disabled={false}
          type="button"
          variant="primary"
        >
          Start Next Workout
        </Button>
      </div>
      <div
        className="input-group workouts-page__filter"
      >
        <select
          className="select-input"
          id="filter-select"
          onChange={[Function]}
          value=""
        >
          <option
            value=""
          >
             All Lifts 
          </option>
          <option
            value="bench"
          >
             
            Bench Press
             
          </option>
          <option
            value="row"
          >
             
            Row
             
          </option>
          <option
            value="squat"
          >
             
            Squat
             
          </option>
          <option
            value="deadlift"
          >
             
            Deadlift
             
          </option>
          <option
            value="overhead"
          >
             
            Overhead Press
             
          </option>
          <option
            value="chinup"
          >
             
            Chin-Ups
             
          </option>
        </select>
        <label
          htmlFor="filter-select"
        >
           Filter Workouts by Lift 
        </label>
      </div>
      <Component
        from={
          Object {
            "opacity": 0,
            "transform": "translate3d(0,-40px,0)",
          }
        }
        items={
          Array [
            Object {
              "bench": Object {
                "result": 2,
                "weight": 85,
              },
              "created": "2020-02-06T18:41:49.124Z",
              "deadlift": Object {
                "result": 1,
                "weight": 106,
              },
              "id": "7a07e348-e0e8-48c8-a60a-3d9fb9929ab5",
              "row": Object {
                "result": 2,
                "weight": 115.5,
              },
            },
            Object {
              "chinup": Object {
                "result": 0,
                "weight": Object {
                  "negatives": 2,
                  "ups": 3,
                  "weight": 0,
                },
              },
              "created": "2020-02-03T21:01:59.832Z",
              "id": "de5f7586-8023-4b3a-bfcf-0ff72fe2c4d8",
              "overhead": Object {
                "result": 2,
                "weight": 107,
              },
              "squat": Object {
                "result": 0,
                "weight": 174,
              },
            },
            Object {
              "chinup": Object {
                "result": 1,
                "weight": Object {
                  "negatives": 1,
                  "ups": 4,
                  "weight": 0,
                },
              },
              "created": "1969-12-08T18:28:12.461Z",
              "id": "1d8ad17a-c62a-4182-8ddb-0df62af33c65",
              "overhead": Object {
                "result": 2,
                "weight": 112,
              },
              "squat": Object {
                "result": 2,
                "weight": 156.6,
              },
            },
          ]
        }
        keys={[Function]}
        to={
          Object {
            "opacity": 1,
            "transform": "translate3d(0,0px,0)",
          }
        }
      >
        <Component />
      </Component>
    </div>
  </div>
</div>
`;

exports[`should render Workouts Page without props 1`] = `
<div
  className="page--with-side-nav"
>
  <SideNav
    path=""
  />
  <div
    className="workouts-page side-nav__page-content"
  >
    <div
      className="side-nav__page-scrim side-nav__page-scrim--hidden"
    />
    <div
      className="workouts-page__list"
    >
      <div
        className="workouts-page__header"
      >
        <h1>
          0 total workouts
        </h1>
        <Button
          clickHandler={[Function]}
          disabled={false}
          type="button"
          variant="primary"
        >
          Start Next Workout
        </Button>
      </div>
      <div
        className="input-group workouts-page__filter"
      >
        <select
          className="select-input"
          id="filter-select"
          onChange={[Function]}
          value=""
        >
          <option
            value=""
          >
             All Lifts 
          </option>
          <option
            value="bench"
          >
             
            Bench Press
             
          </option>
          <option
            value="row"
          >
             
            Row
             
          </option>
          <option
            value="squat"
          >
             
            Squat
             
          </option>
          <option
            value="deadlift"
          >
             
            Deadlift
             
          </option>
          <option
            value="overhead"
          >
             
            Overhead Press
             
          </option>
          <option
            value="chinup"
          >
             
            Chin-Ups
             
          </option>
        </select>
        <label
          htmlFor="filter-select"
        >
           Filter Workouts by Lift 
        </label>
      </div>
      <Component
        from={
          Object {
            "opacity": 0,
            "transform": "translate3d(0,-40px,0)",
          }
        }
        items={Array []}
        keys={[Function]}
        to={
          Object {
            "opacity": 1,
            "transform": "translate3d(0,0px,0)",
          }
        }
      >
        <Component />
      </Component>
    </div>
  </div>
</div>
`;
